src/NLog/Internal/ObjectPools/ReusableBuilderCreator.cs

```diff
@@ -40,8 +40,20 @@ namespace NLog.Internal
     /// </summary>
     internal sealed class ReusableBuilderCreator : ReusableObjectCreator<StringBuilder>
     {
+        private const int InitialCapacity = 128;
+        private const int MaxCapacity = 1024 * 1024 * 5; // 5 MB
+        
         public ReusableBuilderCreator()
-            : base(128, (cap) => new StringBuilder(cap), (sb) => { sb.ClearBuilder(); })
+            : base(initialCapacity: InitialCapacity, maxCapacity: MaxCapacity, (initialCapacity, _) => new StringBuilder(initialCapacity), (initialCapacity, maxCapacity, sb)  =>
+            {
+                if (sb.Length > maxCapacity)
+                {
+                    sb.Remove(0, sb.Length - 1);  // Attempt soft clear that skips re-allocation
+                    
+                    if (sb.Capacity > maxCapacity)
+                        sb.Capacity = maxCapacity;
+                }
```

Comment:
You have now remove the call to `sb.ClearBuilder()`

Reply:
Fixed